lib/inspec/resource.rb

```diff
@@ -3,17 +3,36 @@
 # license: All rights reserved
 # author: Dominik Richter
 # author: Christoph Hartmann
-
+require 'thread'
 require 'inspec/plugins'
 
 module Inspec
   class Resource
+    REGISTRY_LOCK = Mutex.new
```

Comment:
Summary of discussion:

We can probably replace this by introducing a profile-specific class that all library files are evaluated against.